compiler/src/dotty/tools/dotc/reporting/StoreReporter.scala

```diff
@@ -17,7 +17,7 @@ import Diagnostic._
   * - The reporter is not flushed and the message containers capture a
   *   `Context` (about 4MB)
   */
-class StoreReporter(outer: Reporter) extends Reporter {
+class StoreReporter(outer: Reporter = null) extends Reporter {
```

Comment:
I'd suggest remove this default value, as it is not common for `outer` to be `null`. Moreover, this creates more subtleties for the usage of `outer` inside `StoreReporter`.  There is `Reporter.NoReporter`, which can be used while `null` is needed.